### PR TITLE
Added argparse support

### LINKED ISSUE #8: Add CLI arguments
Add argparse support to choose the desired process.

### PR DESCRIPTION
**Pull request by IlanKalendarov**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,3 @@
 credentials.txt
-./.idea
+.idea
```
Comment (LXTreato) [inline .gitignore:2]: This change is unrelated to the issue, I'd suggest moving it to another dedicated pull request or just push it solely straight to main.

File: PyHook.py [hunk 2]
```diff
@@ -4,12 +4,32 @@
...
+ mstsc
+ cmd
+ mobaxterm
+ runas
+ explorer
+ psexec'''
+
+parser = ArgumentParser(epilog=example_text, formatter_class=RawTextHelpFormatter)
+parser.add_argument('-p', '--process', help='desired process to hook', nargs='+', required=True)
+args = parser.parse_args()
```
Comment (LXTreato) [inline PyHook.py]: Consider moving it to a function to avoid loading this code when importing the module.
Then using it in the main function.
```suggestion
def parse_args():
    parser = ArgumentParser(epilog=example_text, formatter_class=RawTextHelpFormatter)
    parser.add_argument('-p', '--process', help='desired process to hook', nargs='+', required=True)
    return parser.parse_args()
```
It's also better to pass the arguments "in-house" instead of using global variable.

File: PyHook.py [hunk 3]
```diff
@@ -4,12 +4,32 @@
...
 from typing import List
 from time import sleep
 import psutil
 import re
 
+example_text = '''example:
+
+ python PyHook.py -p mstsc - Hook into mstsc
+ python PyHook.py -p mstsc, cmd - Hook into mstsc and cmd
+ python PyHook.py -p all - Hook all available processes
```
Comment (LXTreato) [inline PyHook.py]: What happens when you execute it without additional arguments? (e.g: `python PyHook.py`)
You can have it set to use all hooks by default, or have it set to a specific main hook, your choice.
_____
Another thing that's a little bit more nitpicky about the flags.
I think it'll be more intuitive to use them the following way:
```shell
python PyHook.py [-cmxrep]
      ||
      \/
-c, --cmd          enable cmd hook.
-m, --mstsc        enable mstsc hook.
-x, --xterm        enable mobaxterm hook.
-r, --runas        enable runas hook.
-e, --explorer     enable explorerhook.
-p, --psexec       enable psexechook.

No flags           enable all hooks
```
Then you can use this type of flag with argparse as shown [here](https://stackoverflow.com/questions/8259001/python-argparse-command-line-flags-without-arguments#:~:text=As%20you%20have,w%27%2C%20action%3D%27store_true%27)

File: PyHook.py [hunk 4]
```diff
@@ -62,8 +82,33 @@ def on_credential_submit(message, data):
...
+            print("[+] Waiting for explorer")
+            functions.append(explorer.wait_for)
+        elif process == "mstsc":
+            print("[+] Waiting for mstsc")
+            functions.append(rdp.wait_for)
+        elif process == "all":
+            functions.extend((psexec.wait_for, rdp.wait_for, explorer.wait_for, cmd.wait_for, mobaxterm.wait_for,
+                              runas.wait_for))
+        else:
+            print("[-] The process is not available")
```
Comment (LXTreato) [inline PyHook.py]: Consider changing to 
```python
f"[-] A module for the selected process is not available - {name of process}"
```

File: PyHook.py [hunk 5]
```diff
@@ -62,8 +82,33 @@ def on_credential_submit(message, data):
...
+            exit(1)
```
Comment (LXTreato) [inline PyHook.py]: This is a very extensive code for just a simple selection.
Consider using a list to represent the available modules, like so:
```python
import hooks

...

def main():

    ...

    functions = []
    
    hooks_modules = dir(hooks)
    
    for hook_process_name in args.desired_hooks:
        try:
            hook_module = __import__(f"hooks.{hook_process_name}")
            functions.append(hook_module.wait_for)
        except OSError:
            print(f"[-] A module for the selected process is not available - {hook_process_name}")
    
```
This solution will discourage redundant import like this line:
```python
from hooks import rdp, psexec, explorer, cmd, mobaxterm, runas
```

###### *Note - I didn't check the validity of the code so feel free to fix and make adjustments to it*

**Review by LXTreato — CHANGES_REQUESTED** [review]

File: PyHook.py (hunk 2 above)
Comment (IlanKalendarov) [reply PyHook.py]: You said "Consider moving it to a function" I Didn't notice you defined any function in your suggestion

File: PyHook.py (hunk 3 above)
Comment (IlanKalendarov) [reply PyHook.py]: Good idea, Will implement this

File: PyHook.py (hunk 2 above)
Comment (LXTreato) [reply PyHook.py]: line 29 in my suggestion

File: PyHook.py (hunk 4 above)
Comment (IlanKalendarov) [reply PyHook.py]: 👍

File: PyHook.py (hunk 5 above)
Comment (IlanKalendarov) [reply PyHook.py]: Will do

File: PyHook.py (hunk 2 above)
Comment (IlanKalendarov) [reply PyHook.py]: image

I guess it's a bug in the GitHub mobile app 🤷‍♂️
